garage/contrib/exp/checkpointers/disk_checkpointer.py

```diff
@@ -0,0 +1,161 @@
+import pickle as pkl
+from os import listdir, path, remove, makedirs, rmdir
+
+from garage.contrib.exp.checkpointers.checkpointer import Checkpointer
+from garage.contrib.exp.checkpointers.checkpointer import cat_for_fname
+from garage.contrib.exp.checkpointers.checkpointer import get_now_timestamp
+from garage.contrib.exp.checkpointers.checkpointer import get_timestamp
+
+
+class DiskCheckpointer(Checkpointer):
+    def __init__(self, exp_dir, prefix, resume=True):
+        super(DiskCheckpointer, self).__init__(prefix, resume)
+        self.exp_dir = exp_dir
+
+    def load(self, **kwargs):
+        """Load checkpoint from disk if there exists.
+
+        Load checkpoint from exp_dir directory.
+
+        If there's no saved checkpoint, create an initial checkpoint instead.
+        This usually happens when an experiment is run for the first time
+        and the initial states will be saved.
+
+        A checkpoints might consist of several files, each is named
+        in the format of [prefix_]timestamp_object.pkl and corresponds
+        to a named entry in kwargs.
+
+        Args:
+            **kwargs: Objects to save.
+                The name of argument is used to name checkpoint file.
+
+        Returns:
+            dict: restored objects from disk.
+
+        """
+        latest_checkpoint, _ = self._get_latest_checkpoint(kwargs.keys(), dry=True)
+        if not latest_checkpoint or not self.resume:
+            self.save(**kwargs)
+            return kwargs
+        else:
+            return self._load(**kwargs)
+
+    def save(self, **kwargs):
```

Comment:
i think concrete implementations should have (only) concrete arguments.

Reply:
I see what's going on here.

You are using kwargs as a dict to allow the user to avoid the boilerplate of constructing one themselves. Perhaps.

@krzentner what do you think of this interface? what if instead all you can give is something pickleable?